pyproject.toml

```diff
@@ -1,7 +1,66 @@
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-# Testing tools configuration
+[tool.poetry]
+# Charm is not packed as a standard Python package; this information is not used
+name = "charm"
+version = "0.1.0"
+description = ""
+authors = []
+
+[tool.poetry.dependencies]
+python = "^3.10"
+ops = "^2.4.1"
+tenacity = "^8.2.2"
+boto3 = "^1.28.7"
+# TODO: Remove insecure dependency https://github.com/canonical/mysql-operator/issues/246
+pyopenssl = "^23.2.0"
+
+[tool.poetry.group.charm-libs.dependencies]
+# data_platform_libs/v0/data_interfaces.py
+ops = ">=2.0.0"
+# grafana_agent/v0/cos_agent.py
+cosl = "*"
+pydantic = "<2"
+# tls_certificates_interface/v1/tls_certificates.py
+cryptography = "*"
+jsonschema = "*"
```

Comment:
Is this group manually maintained? If so, I think people will only update it when things break, so I don't think it will warn us of dependency conflicts.